go/fn/object_test.go

```diff
@@ -10,6 +10,78 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+type GVKTest struct {
+	group   string
+	version string
+	kind    string
+	resultA bool
+	resultB bool
+	resultC bool
+}
+
+func TestGVK(t *testing.T) {
+	input := []byte(`
+apiVersion: apps/v3
+kind: StatefulSet
+metadata:
+  name: my-config
+spec:
+  volumeClaimTemplates:
+    - metadata:
+        labels:
+          testkey: testvalue
+`)
+
+	input_noGroup := []byte(`
+apiVersion: v3
+kind: StatefulSet
+metadata:
+  name: my-config
+spec:
+  volumeClaimTemplates:
+    - metadata:
+        labels:
+          testkey: testvalue
+`)
+
+	input_differentKind := []byte(`
```

Comment:
I think it would be better to put add "input" field to the `GVKTest struct`, and then pass them to the test cases like

```
    GVKTest {
			input:   input_differentKind,
			group:   "apps",
			version: "",
			kind:    "StatefulSet",
			resultA: true,
			resultB: true,
			resultC: true,
    }
```

That way, each GVKTest is exactly one test case and it is easier to see what is being tested